core/src/sumeragi/mod.rs

```diff
@@ -188,42 +178,50 @@ impl Sumeragi {
         block_hash: HashOf<VersionedCommittedBlock>,
     ) -> Vec<VersionedCommittedBlock> {
         self.internal
-            .wsv
+            .wsv_for_public_use
             .lock()
             .unwrap()
             .blocks_after_hash(block_hash)
     }
 
     pub fn get_random_peer_for_block_sync(&self) -> Option<Peer> {
-        use rand::{prelude::SliceRandom, SeedableRng};
+        use rand::{prelude::SliceRandom, RngCore, SeedableRng};
 
         let rng = &mut rand::rngs::StdRng::from_entropy();
-        self.internal
-            .wsv
+        let mut vec = self
```

Comment:
I will take your suggestion. But the reason I tend to split things up is because I'm dividing up the problem in my head. Long chains of method calls on top of each other are harder to debug and harder to understand than discrete imperitive steps. One is using `object.method1().method2().method3()...` as a replacement for
```
procedure1(object);
procedure2(object);
procedure3(object);
```
Both have mutability, except in the former it is hidden and never declared openly. I think the later method is better because it's more powerful and allows you to insert code to inspect the state of `object` in between the operations.

Reply:
"I don't see why sorting couldn't have been done before collect()". Could you provide code of such a thing? I applied your suggestion without thinking much and I don't think it is possible. You must have all the data to be able to sort it. You can't sort on an iterator.